libevmasm/Assembly.cpp

```diff
@@ -1657,7 +1657,16 @@ LinkerObject const& Assembly::assembleEOF() const
 	}
 
 	for (auto i: referencedSubIds)
-		ret.bytecode += m_subs[i]->assemble().bytecode;
+	{
+		size_t const subAssemblyPostionInParentObject = ret.bytecode.size();
+		auto const& subAssemblyLinkerObject = m_subs[i]->assemble();
+		// Append subassembly bytecode to the parent assembly result bytecode
+		ret.bytecode += subAssemblyLinkerObject.bytecode;
+		// Add subassembly link references to parent linker object.
+		// Offset accordingly to subassembly position in parent object bytecode
+		for (auto const& [subAssemblyLinkRefPosition, linkRef]: subAssemblyLinkerObject.linkReferences)
+			ret.linkReferences[subAssemblyPostionInParentObject + subAssemblyLinkRefPosition] = linkRef;
+	}
```

Comment:
I see that this still does not include the subassembly deduplication we introduced for legacy recently (#13804). I guess we can't do it the same way for EOF anyway since sub-assembly IDs are used as immediate arguments in some opcodes. We'd have to de-duplicate it at a higher level - in Yul or evmasm optimizer.